src/main/java/group1/mips_simulator/components/cpuParts/RegisterFile.java

```diff
@@ -14,46 +14,53 @@ public class RegisterFile {
     Design note: when hardware is taped out and manufactured, there's no way to add/remove registers.
     that's why im replacing the Vector (used in MipsComputer, which im renaming to Computer) with a preset array
      */
-    Register[] gpr = new Register[4];   //General Purpose Reg: R0-R3
-    Register[] ixr = new Register[3];   //Index Reg: X1-X3.
+    Register[] gpr = new Register[GPR_COUNT];   //General Purpose Reg: R0-R3
+    Register[] ixr = new Register[IXR_COUNT];   //Index Reg: X1-X3.
     Register pc = new Register();       //Prog Counter Reg : address of next instruction to be executed
     Register cc = new Register();       //Condition Code Reg : set when arith/logical operations executed
-                                        // ONZE (Overflow, uNderflow, divZero, Equalornot)
-                                // (only one i dont understand is "equal or not", is that saying, set flat to 1 if equal
-                                // and set to 0 if not?)
+    // ONZE (Overflow, uNderflow, divZero, Equalornot)
+    // (only one i dont understand is "equal or not", is that saying, set flat to 1 if equal
+    // and set to 0 if not?)
```

Comment:
Let me know if you wanna talk about this. Basically, the Condition Code is a 4 bit register. The 0th (left most) bit is the overflow bit. 1 => overflow has happened in the previous command, 0=> no overflow.
the 1st bit (to the right) means there was an Underflow. The 2nd bit is a divid by zero signal and the last bit is the "Equal to 0" bit used for conditional jumps.